### PR TITLE
Fix issue 6167 for Find-DbaDbUnusedIndex to include index size and row count

### LINKED ISSUE #6167: Find-DbaDbUnusedIndex show index size and row count
I would like Find-DbaDbUnusedIndex to also show the size of the indexes and the row counts. Like the find duplicate index command does.

### PR DESCRIPTION
**Pull request by lancasteradam**
Fix #6167 for Find-DbaDbUnusedIndex to include index size and row count.

1. Added in the index size and row count columns to the result set. The compression description is also added for SQL 2008 or higher.
2. Added 3 new optional params to allow callers to more flexibly define 'unused' indexes.
3. Added a warning message if the database provided by the caller does not exist.
4. Added 2 integration tests.

## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6167 )
 - [x] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 

### Purpose
 
Fixing issue #6167

### Approach

New columns are added into the returned result set.

### Commands to test

Find-DbaDbUnusedIndex -SqlInstance "enter name here" -Database "enter name here" -IgnoreUptime -UserSeeksLessThan 10 -UserScansLessThan 10 -UserLookupsLessThan 10

The pester test now contains integration tests and has sample command invocations.

### CODE REVIEW

File: functions/Find-DbaDbUnusedIndex.ps1 [hunk 1]
```diff
@@ -35,6 +35,30 @@ function Find-DbaDbUnusedIndex {
...
+
+    .PARAMETER UserScansLessThan
+        Specify a custom threshold for user scans. The default for this parameter is 1.
+        The valid values are between 1 and 1000000 to provide flexibility on the definition of 'unused' indexes.
+        Note: The resulting WHERE clause uses the AND operator:
+        user_seeks < $UserSeeksLessThan
+        AND user_scans < $UserScansLessThan
+        AND user_lookups < $UserLookupsLessThan
+
+    .PARAMETER UserLookupsLessThan
```
Comment (wsmelton) [inline functions/Find-DbaDbUnusedIndex.ps1:54]: If we are going to assume these will be the "max limit" and the comparison will be less than said value...then we can drop the `LessThan` from the parameter naming and simply document it.

So we can just keep them as `UserSeeks` `UserScans` `UserLookups`, or really we can just do `Seeks` `Scans` and `Lookups`.  It is well known they are user based stats, so do we really need the "user" in the parameter name, @potatoqualitee ?

File: functions/Find-DbaDbUnusedIndex.ps1 [hunk 2]
```diff
@@ -72,11 +96,40 @@ function Find-DbaDbUnusedIndex {
         [switch]$IgnoreUptime,
         [Parameter(ValueFromPipeline)]
         [Microsoft.SqlServer.Management.Smo.Database[]]$InputObject,
-        [switch]$EnableException
+        [switch]$EnableException,
+        [ValidateRange(1, 1000000)][Int]$UserSeeksLessThan = 1,
+        [ValidateRange(1, 1000000)][Int]$UserScansLessThan = 1,
+        [ValidateRange(1, 1000000)][Int]$UserLookupsLessThan = 1
```
Comment (wsmelton) [inline functions/Find-DbaDbUnusedIndex.ps1:102]: Once we finalize the parameter name, pleas add examples showing their use.

Comment (wsmelton) [reply functions/Find-DbaDbUnusedIndex.ps1:102]: `[int]` instead of `[Int]` as well please 😉

File: functions/Find-DbaDbUnusedIndex.ps1 [hunk 3]
```diff
@@ -72,11 +96,40 @@ function Find-DbaDbUnusedIndex {
         [switch]$IgnoreUptime,
         [Parameter(ValueFromPipeline)]
         [Microsoft.SqlServer.Management.Smo.Database[]]$InputObject,
-        [switch]$EnableException
+        [switch]$EnableException,
```
Comment (wsmelton) [inline functions/Find-DbaDbUnusedIndex.ps1:99]: Please place all new parameters prior to the `$InputObject` parameter in this param block.

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 [hunk 4]
```diff
@@ -1,19 +1,136 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+$script:UnusedIndexCommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
```
Comment (wsmelton) [inline tests/Find-DbaDbUnusedIndex.Tests.ps1:1]: Please revert this change.

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 [hunk 5]
```diff
@@ -1,19 +1,136 @@
...
\ No newline at end of file
+#>
+
+
+Describe "$script:UnusedIndexCommandName Integration Tests" -Tags "IntegrationTests" {
+    Context "Verify basics of the Find-DbaDbUnusedIndex command" {
+        BeforeAll {
+            $server1 = Connect-DbaInstance -SqlInstance $script:instance1
+            $server2 = Connect-DbaInstance -SqlInstance $script:instance2
+            $server3 = Connect-DbaInstance -SqlInstance $script:instance3
```
Comment (wsmelton) [inline tests/Find-DbaDbUnusedIndex.Tests.ps1:28]: Please only perform the test on instance3.

File: functions/Find-DbaDbUnusedIndex.ps1 (hunk 2 above)
Comment (lancasteradam) [reply functions/Find-DbaDbUnusedIndex.ps1:102]: ok, sounds good

File: functions/Find-DbaDbUnusedIndex.ps1 (hunk 3 above)
Comment (lancasteradam) [reply functions/Find-DbaDbUnusedIndex.ps1:99]: ok, will do

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 (hunk 4 above)
Comment (lancasteradam) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1:1]: Ok, I can revert that change. For some reason on my machine the scope of $CommandName causes the Get-Command on line 8 to fail because $CommandName is null in that block. When I had checked the Pester documentation it seemed to suggest that using the script scope was needed.

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 (hunk 5 above)
Comment (lancasteradam) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1:28]: Sure, np

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 (hunk 4 above)
Comment (wsmelton) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1]: Hmm. Was that referenced in our documentation?

General process I use that tends to with is importing the module psd1 and pester into the session. Then invoke the tests script.

Let me know if that was not understood or documented wrong and I'll get that updated.

Comment (lancasteradam) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1]: Thanks, that was the tip I needed. It seems running the test using Visual Studio Code was the issue with the variable scope since it starts the execution from a specific line number. I'm able to move forward now and I'm looking into an AppVeyor error that occurred in part 1 of the default scenario. 

Is there a process for running our own AppVeyor before creating a PR? I can help edit the instructional text and test it.

Comment (wsmelton) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1]: A commit will trigger another test run on appveyor. You can format your commit as `(do find-dbadbunusedindex)` and that will trigger only the test for that function to run. Saves a bit of time until I can with out optimizing our test framework again.

Running the test locally though in your own environment our in a container should suffice in most situations.

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 (hunk 5 above)
Comment (lancasteradam) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1]: There seems to be an issue with using instance3. The [default scenario](https://ci.appveyor.com/project/sqlcollaborative/dbatools/builds/34725387/job/in72ya3mndc7e8p9/artifacts) in AppVeyor is getting selected and then at runtime there is an instance failure when trying to connect to the sql2017 instance. Is there an issue with the dynamic scenario selection? If I use [instance2](https://ci.appveyor.com/project/sqlcollaborative/dbatools/builds/34726052/job/4ebaf7299tdf8uud) the test works fine.

**Comment by wsmelton** [comment]
@lancasteradam in order to have GitHub acknowledge your contribution we have to merge the PR, but the commits for this one have gotten a bit wild being that issues with Appveyor were being worked out.

In that, if you will, once you get everything squared away please close this PR. Create a new branch and add your changes then create a PR against that branch. This will help keep our commit history a bit cleaner for your contributions.

File: functions/Find-DbaDbUnusedIndex.ps1 (hunk 1 above)
Comment (lancasteradam) [reply functions/Find-DbaDbUnusedIndex.ps1:54]: I can change the param names to Seeks, Scans, and Lookups. @potatoqualitee, do you have a strong opinion either way on the names for these command line params?

File: tests/Find-DbaDbUnusedIndex.Tests.ps1 (hunk 5 above)
Comment (wsmelton) [reply tests/Find-DbaDbUnusedIndex.Tests.ps1]: Your last run shows the test running under the excepted scenario for sql2016_sql2017

**Comment by lancasteradam** [comment]
@wsmelton , no problem I'll mark this PR as closed. The new PR is ready for review here: https://github.com/sqlcollaborative/dbatools/pull/6748
